pipeline/relation_embedder/path_concatenator/src/test/scala/weco/pipeline/path_concatenator/PathsServiceTest.scala

```diff
@@ -90,11 +92,30 @@ class PathsServiceTest
         work(path = "a/b/c/d/e"),
         work(path = "e/f/g/h/i")
       )
-      withLocalMergedWorksIndex { index =>
-        insertIntoElasticsearch(index, works: _*)
-        whenReady(service(index).getParentPath("e/f/g/h/i")) {
-          _ shouldBe Some("a/b/c/d/e")
-        }
+      withLocalMergedWorksIndex {
+        index =>
+          insertIntoElasticsearch(index, works: _*)
+          whenReady(service(index).getParentPath("e/f/g/h/i")) {
+            _ shouldBe Some("a/b/c/d/e")
+          }
+      }
+    }
+
+    it("does not add a parent to a root-only path") {
+      // When the requested path consists of a single node, it is the root of its own tree,
+      // and cannot have a parent.
+      // An earlier iteration of this service would get confused by the trailing slash in
+      // a path in a different tree.
+      val works: List[Work[Merged]] = List(
+        work(path = "parent"),
+        work(path = "a/completely/different/tree/")
+      )
+      withLocalMergedWorksIndex {
+        index =>
+          insertIntoElasticsearch(index, works: _*)
+          whenReady(service(index).getParentPath("parent")) {
+            _ shouldBe empty
+          }
```

Comment:
This is the only actual addition to this file.  The rest is IntelliJ deciding it didn't like the formatting.  Sorry about that.